error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:31:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
31 |     is_unpin::<Foo<(), ()>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo<(), ()>`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:32:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
32 |     is_unpin::<Foo<PhantomPinned, ()>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo<std::marker::PhantomPinned, ()>`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:33:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
33 |     is_unpin::<Foo<(), PhantomPinned>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo<(), std::marker::PhantomPinned>`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:34:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
34 |     is_unpin::<Foo<PhantomPinned, PhantomPinned>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Foo<std::marker::PhantomPinned, std::marker::PhantomPinned>`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:36:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
36 |     is_unpin::<TrivialBounds>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `TrivialBounds`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:38:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
38 |     is_unpin::<Bar<'_, (), ()>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Bar<'_, (), ()>`

error[E0277]: `std::marker::PhantomPinned` cannot be unpinned
  --> $DIR/assert-not-unpin.rs:39:5
   |
28 | fn is_unpin<T: Unpin>() {}
   |                ----- required by this bound in `is_unpin`
...
39 |     is_unpin::<Bar<'_, PhantomPinned, PhantomPinned>>(); //~ ERROR E0277
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ within `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`, the trait `std::marker::Unpin` is not implemented for `std::marker::PhantomPinned`
   |
   = note: required because it appears within the type `pin_project::__private::Wrapper<'_, std::marker::PhantomPinned>`
   = note: required because of the requirements on the impl of `std::marker::Unpin` for `Bar<'_, std::marker::PhantomPinned, std::marker::PhantomPinned>`
