error: unexpected token: ()
 --> $DIR/invalid.rs:6:14
  |
6 |         #[pin()] //~ ERROR unexpected token
  |              ^^

error: unexpected token: (foo)
  --> $DIR/invalid.rs:11:29
   |
11 |     struct TupleStruct(#[pin(foo)] ()); //~ ERROR unexpected token
   |                             ^^^^^

error: unexpected token: (foo)
  --> $DIR/invalid.rs:15:16
   |
15 |         V(#[pin(foo)] ()), //~ ERROR unexpected token
   |                ^^^^^

error: unexpected token: (foo)
  --> $DIR/invalid.rs:21:18
   |
21 |             #[pin(foo)] //~ ERROR unexpected token
   |                  ^^^^^

error: duplicate #[pin] attribute
  --> $DIR/invalid.rs:33:9
   |
33 |         #[pin] //~ ERROR duplicate #[pin] attribute
   |         ^^^^^^

error: duplicate #[pin] attribute
  --> $DIR/invalid.rs:40:9
   |
40 |         #[pin]
   |         ^^^^^^

error: duplicate #[pin] attribute
  --> $DIR/invalid.rs:49:13
   |
49 |             #[pin]
   |             ^^^^^^

error: duplicate #[pin] attribute
  --> $DIR/invalid.rs:59:13
   |
59 |             #[pin] //~ ERROR duplicate #[pin] attribute
   |             ^^^^^^

error: #[pin] attribute may only be used on fields of structs or variants
  --> $DIR/invalid.rs:69:5
   |
69 |     #[pin] //~ ERROR may only be used on fields of structs or variants
   |     ^^^^^^

error: #[pin] attribute may only be used on fields of structs or variants
  --> $DIR/invalid.rs:77:9
   |
77 |         #[pin] //~ ERROR may only be used on fields of structs or variants
   |         ^^^^^^

error: #[pin] attribute may only be used on fields of structs or variants
  --> $DIR/invalid.rs:82:5
   |
82 |     #[pin] //~ ERROR may only be used on fields of structs or variants
   |     ^^^^^^

error: expected identifier
  --> $DIR/invalid.rs:91:31
   |
91 |     #[pin_project(UnsafeUnpin,,)] //~ ERROR expected identifier
   |                               ^

error: unexpected argument: Foo
  --> $DIR/invalid.rs:94:19
   |
94 |     #[pin_project(Foo)] //~ ERROR unexpected argument
   |                   ^^^

error: expected identifier
  --> $DIR/invalid.rs:97:19
   |
97 |     #[pin_project(,UnsafeUnpin)] //~ ERROR expected identifier
   |                   ^

error: duplicate `PinnedDrop` argument
   --> $DIR/invalid.rs:103:31
    |
103 |     #[pin_project(PinnedDrop, PinnedDrop)] //~ ERROR duplicate `PinnedDrop` argument
    |                               ^^^^^^^^^^

error: duplicate `Replace` argument
   --> $DIR/invalid.rs:106:28
    |
106 |     #[pin_project(Replace, Replace)] //~ ERROR duplicate `Replace` argument
    |                            ^^^^^^^

error: duplicate `UnsafeUnpin` argument
   --> $DIR/invalid.rs:109:32
    |
109 |     #[pin_project(UnsafeUnpin, UnsafeUnpin)] //~ ERROR duplicate `UnsafeUnpin` argument
    |                                ^^^^^^^^^^^

error: duplicate `!Unpin` argument
   --> $DIR/invalid.rs:112:27
    |
112 |     #[pin_project(!Unpin, !Unpin)] //~ ERROR duplicate `!Unpin` argument
    |                           ^^^^^^

error: duplicate `UnsafeUnpin` argument
   --> $DIR/invalid.rs:115:44
    |
115 |     #[pin_project(PinnedDrop, UnsafeUnpin, UnsafeUnpin)] //~ ERROR duplicate `UnsafeUnpin` argument
    |                                            ^^^^^^^^^^^

error: duplicate `PinnedDrop` argument
   --> $DIR/invalid.rs:118:44
    |
118 |     #[pin_project(PinnedDrop, UnsafeUnpin, PinnedDrop, UnsafeUnpin)] //~ ERROR duplicate `PinnedDrop` argument
    |                                            ^^^^^^^^^^

error: duplicate `project` argument
   --> $DIR/invalid.rs:121:32
    |
121 |     #[pin_project(project = A, project = B)] //~ ERROR duplicate `project` argument
    |                                ^^^^^^^^^^^

error: duplicate `project` argument
   --> $DIR/invalid.rs:124:49
    |
124 |     #[pin_project(project = A, project_ref = A, project = B)] //~ ERROR duplicate `project` argument
    |                                                 ^^^^^^^^^^^

error: duplicate `project_ref` argument
   --> $DIR/invalid.rs:127:36
    |
127 |     #[pin_project(project_ref = A, project_ref = B)] //~ ERROR duplicate `project_ref` argument
    |                                    ^^^^^^^^^^^^^^^

error: duplicate `project_replace` argument
   --> $DIR/invalid.rs:130:40
    |
130 |     #[pin_project(project_replace = A, project_replace = B)] //~ ERROR duplicate `project_replace` argument
    |                                        ^^^^^^^^^^^^^^^^^^^

error: `project_replace` argument can only be used together with `Replace` argument
   --> $DIR/invalid.rs:133:19
    |
133 |     #[pin_project(project_replace = A)] //~ ERROR `project_replace` argument can only be used together with `Replace` argument
    |                   ^^^^^^^^^^^^^^^^^^^

error: arguments `PinnedDrop` and `Replace` are mutually exclusive
   --> $DIR/invalid.rs:136:19
    |
136 |     #[pin_project(PinnedDrop, Replace)] //~ ERROR arguments `PinnedDrop` and `Replace` are mutually exclusive
    |                   ^^^^^^^^^^

error: arguments `PinnedDrop` and `Replace` are mutually exclusive
   --> $DIR/invalid.rs:139:41
    |
139 |     #[pin_project(Replace, UnsafeUnpin, PinnedDrop)] //~ ERROR arguments `PinnedDrop` and `Replace` are mutually exclusive
    |                                         ^^^^^^^^^^

error: arguments `UnsafeUnpin` and `!Unpin` are mutually exclusive
   --> $DIR/invalid.rs:142:19
    |
142 |     #[pin_project(UnsafeUnpin, !Unpin)] //~ ERROR arguments `UnsafeUnpin` and `!Unpin` are mutually exclusive
    |                   ^^^^^^^^^^^

error: arguments `UnsafeUnpin` and `!Unpin` are mutually exclusive
   --> $DIR/invalid.rs:145:39
    |
145 |     #[pin_project(!Unpin, PinnedDrop, UnsafeUnpin)] //~ ERROR arguments `UnsafeUnpin` and `!Unpin` are mutually exclusive
    |                                       ^^^^^^^^^^^

error: expected `!Unpin`, found `!`
   --> $DIR/invalid.rs:148:19
    |
148 |     #[pin_project(!)] //~ ERROR expected `!Unpin`, found `!`
    |                   ^

error: unexpected argument: Unpin
   --> $DIR/invalid.rs:151:19
    |
151 |     #[pin_project(Unpin)] //~ ERROR unexpected argument
    |                   ^^^^^

error: expected `project = <identifier>`, found `project`
   --> $DIR/invalid.rs:154:19
    |
154 |     #[pin_project(project)] //~ ERROR expected `project = <identifier>`, found `project`
    |                   ^^^^^^^

error: expected `project = <identifier>`, found `project =`
   --> $DIR/invalid.rs:157:19
    |
157 |     #[pin_project(project = )] //~ ERROR expected `project = <identifier>`, found `project =`
    |                   ^^^^^^^^^

error: expected `project_ref = <identifier>`, found `project_ref`
   --> $DIR/invalid.rs:160:19
    |
160 |     #[pin_project(project_ref)] //~ ERROR expected `project_ref = <identifier>`, found `project_ref`
    |                   ^^^^^^^^^^^

error: expected `project_ref = <identifier>`, found `project_ref =`
   --> $DIR/invalid.rs:163:19
    |
163 |     #[pin_project(project_ref = )] //~ ERROR expected `project_ref = <identifier>`, found `project_ref =`
    |                   ^^^^^^^^^^^^^

error: expected `project_replace = <identifier>`, found `project_replace`
   --> $DIR/invalid.rs:166:19
    |
166 |     #[pin_project(project_replace)] //~ ERROR expected `project_replace = <identifier>`, found `project_replace`
    |                   ^^^^^^^^^^^^^^^

error: expected `project_replace = <identifier>`, found `project_replace =`
   --> $DIR/invalid.rs:169:19
    |
169 |     #[pin_project(project_replace = )] //~ ERROR expected `project_replace = <identifier>`, found `project_replace =`
    |                   ^^^^^^^^^^^^^^^^^

error: duplicate #[pin_project] attribute
   --> $DIR/invalid.rs:177:5
    |
177 |     #[pin_project] //~ ERROR duplicate #[pin_project] attribute
    |     ^^^^^^^^^^^^^^

error: #[pin_project] attribute may not be used on structs with zero fields
   --> $DIR/invalid.rs:185:19
    |
185 |     struct Struct {} //~ ERROR may not be used on structs with zero fields
    |                   ^^

error: #[pin_project] attribute may not be used on structs with zero fields
   --> $DIR/invalid.rs:188:23
    |
188 |     struct TupleStruct(); //~ ERROR may not be used on structs with zero fields
    |                       ^^

error: #[pin_project] attribute may not be used on structs with zero fields
   --> $DIR/invalid.rs:191:12
    |
191 |     struct UnitStruct; //~ ERROR may not be used on structs with zero fields
    |            ^^^^^^^^^^

error: #[pin_project] attribute may not be used on enums without variants
   --> $DIR/invalid.rs:194:20
    |
194 |     enum EnumEmpty {} //~ ERROR may not be used on enums without variants
    |                    ^^

error: #[pin_project] attribute may not be used on enums with discriminants
   --> $DIR/invalid.rs:198:13
    |
198 |         V = 2, //~ ERROR may not be used on enums with discriminants
    |             ^

error: #[pin_project] attribute may not be used on enums with zero fields
   --> $DIR/invalid.rs:203:9
    |
203 | /         Unit, //~ ERROR may not be used on enums with zero fields
204 | |         Tuple(),
205 | |         Struct {},
    | |__________________^

error: #[pin_project] attribute may only be used on structs or enums
   --> $DIR/invalid.rs:209:5
    |
209 | /     union Union {
210 | |         //~^ ERROR may only be used on structs or enums
211 | |         f: (),
212 | |     }
    | |_____^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
   --> $DIR/invalid.rs:220:12
    |
220 |     #[repr(packed)]
    |            ^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
   --> $DIR/invalid.rs:224:12
    |
224 |     #[repr(packed)]
    |            ^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
   --> $DIR/invalid.rs:228:12
    |
228 |     #[repr(packed)]
    |            ^^^^^^
