error: `self` parameter is only allowed in associated functions
  --> $DIR/self.rs:17:26
   |
17 |                     fn f(self: ()) {} //~ ERROR `self` parameter is only allowed in associated functions
   |                          ^^^^ not semantically valid as function parameter
...
20 |             t!();
   |             ----- in this macro invocation
   |
   = note: associated functions are those in `impl` or `trait` definitions
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0434]: can't capture dynamic environment in a fn item
  --> $DIR/self.rs:15:29
   |
15 |                     let _ = self; //~ ERROR E0434
   |                             ^^^^
...
20 |             t!();
   |             ----- in this macro invocation
   |
   = help: use the `|| { ... }` closure form instead
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0423]: expected value, found struct `S`
  --> $DIR/self.rs:38:27
   |
30 | /     pub struct S {
31 | |         x: (),
32 | |     }
   | |_____- `S` defined here
...
38 |               let _: Self = Self; //~ ERROR E0423
   |                             ^^^^ did you mean `S { /* fields */ }`?

error[E0308]: mismatched types
  --> $DIR/self.rs:37:25
   |
37 |             let _: () = self; //~ ERROR E0308
   |                    --   ^^^^ expected `()`, found struct `std::pin::Pin`
   |                    |
   |                    expected due to this
   |
   = note: expected unit type `()`
                 found struct `std::pin::Pin<&mut self_span::S>`
